string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,128 @@
+      } else if (output[output.length - 2] === ':') {
+        capitalWord += word[0].toUpperCase();
+        for (var g = 1; g < word.length; g++) {
+          capitalWord += word[g];
+        }
+        if (capitalWord.toLowerCase() === 'javascript') {
+          capitalWord = 'JavaScript';
+        } else if (capitalWord.toLowerCase() === 'web') {
+          capitalWord = 'Web';
+        } else if (capitalWord.toLowerCase() === 'api') {
+          capitalWord = 'API';
+        }
+        capitalWord += title[i];
+        output += capitalWord;
+        capitalWord = '';
+      } else if (word.length > 3 && title[i] === ' ') {
+        capitalWord += word[0].toUpperCase();
+        for (var c = 1; c < word.length; c++) {
+          capitalWord += word[c];
+        }
+        if (capitalWord.toLowerCase() === 'javascript') {
+          capitalWord = 'JavaScript';
+        } else if (capitalWord.toLowerCase() === 'web') {
+          capitalWord = 'Web';
+        } else if (capitalWord.toLowerCase() === 'api') {
+          capitalWord = 'API';
+        } else if (capitalWord.toLowerCase() === 'javascript:') {
+          capitalWord = 'JavaScript:';
+        }
+        capitalWord += title[i];
+        output += capitalWord;
+        capitalWord = '';
+      } else if (title[i] === '-') {
+        capitalWord += word[0].toUpperCase();
+        for (var d = 1; d < word.length; d++) {
+          capitalWord += word[d];
+        }
+        output += capitalWord;
+        capitalWord = '';
+        if (capitalWord.toLowerCase() === 'javascript') {
+          capitalWord = 'JavaScript';
+        } else if (capitalWord.toLowerCase() === 'web') {
+          capitalWord = 'Web';
+        } else if (capitalWord.toLowerCase() === 'api') {
+          capitalWord = 'API';
+        }
+        capitalWord += title[i];
+        output += capitalWord;
+        capitalWord = '';
+      } else {
+        if (word.toLowerCase() === 'javascript') {
+          word = 'JavaScript';
+        } else if (word.toLowerCase() === 'web') {
+          word = 'Web';
+        } else if (word.toLowerCase() === 'api') {
+          word = 'API';
+        }
+        word += title[i];
+        output += word;
+      }
+      word = '';
+    } else {
+      word += title[i];
+    }
+  }
+  if (word.length > 3) {
+    capitalWord += word[0].toUpperCase();
+    for (var l = 1; l < word.length; l++) {
+      capitalWord += word[l];
+    }
+    if (capitalWord.toLowerCase() === 'javascript') {
+      capitalWord = 'JavaScript';
+    } else if (capitalWord.toLowerCase() === 'web') {
+      capitalWord = 'Web';
+    } else if (capitalWord.toLowerCase() === 'api') {
+      capitalWord = 'API';
+    }
+    output += capitalWord;
+  } else {
+    if (word.toLowerCase() === 'javascript') {
+      word = 'JavaScript';
+    } else if (word.toLowerCase() === 'web') {
+      word = 'Web';
+    } else if (word.toLowerCase() === 'api') {
+      word = 'API';
+    }
+    output += word;
+  }
```

Comment:
There is a **_lot_** of repetition here. You have work for the "javascript", "web", and "API" words 8 times in this function. Try to revise this in a way where any action only needs to be stated in the code once.